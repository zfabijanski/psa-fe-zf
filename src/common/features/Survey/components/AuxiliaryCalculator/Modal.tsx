import { FC } from "react";
import { createPortal } from "react-dom";
import { FocusOn } from "react-focus-on";
import styled from "styled-components";
import { Backdrop } from "components/UI/Backdrop";
import ExpensesCalculator from "./ExpensesCalculator";
import { IRow as IExpensesRow } from "./ExpensesCalculator/types";
import HealthProblemsCalculator from "./HealthProblemsCalculator";
import { IHealthProblemData } from "./HealthProblemsCalculator/types";
import NeedsCalculator from "./NeedsCalculator";
import { IRow as INeedsRow } from "./NeedsCalculator/types";
import { Calculator, CalculatorData } from "./types";

const ModalBody = styled.div<IDisplay>`
  display: ${({ display }) => display};
  position: fixed;
  z-index: 310;
  top: 0;
  left: 15%;
  justify-content: center;
  align-items: center;
  height: 100%;
`;

const Container = styled.div`
  height: 100%;
  width: 100%;
  z-index: 320;
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 0px;
  left: 0px;
`;

interface IDisplay {
  display: string;
}
interface IProps {
  isEnabled: boolean;
  type: Calculator;
  data: CalculatorData;
  onSave: (sum: number) => void;
  onCancel: () => void;
  validationPattern?: string | RegExp;
}

const Modal: FC<IProps> = (props) => {
  return (
    <div>
      {createPortal(
        <FocusOn enabled={props.isEnabled}>
          <ModalBody display={props.isEnabled ? "flex" : "none"}>
            <Container>
              {props.type === "needs" && (
                <NeedsCalculator
                  needs={props.data as INeedsRow[]}
                  onSave={props.onSave}
                  onCancel={props.onCancel}
                />
              )}
              {props.type === "expenses" && (
                <ExpensesCalculator
                  data={props.data as IExpensesRow[]}
                  onSave={props.onSave}
                  onCancel={props.onCancel}
                  validationPattern={props.validationPattern || ""}
                />
              )}
              {props.type === "healthProblems" && (
                <HealthProblemsCalculator
                  data={props.data as IHealthProblemData[]}
                  onSave={props.onSave}
                  onCancel={props.onCancel}
                />
              )}
            </Container>
            <Backdrop />
          </ModalBody>
        </FocusOn>,
        document.getElementById("modal") || document.body
      )}
    </div>
  );
};

export default Modal;
