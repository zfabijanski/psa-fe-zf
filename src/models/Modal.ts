import { ReactNode } from "react";
import { TIconType } from "../components/UI/Icon";
import { TranslationValue } from "utils/types";

export enum ButtonTypes {
  cancel = "cancel",
  confirm = "confirm",
}

export enum ModalTypes {
  default = "default",
  danger = "danger",
}

export interface IModalData {
  modalIllustration?: TIconType;
  modalTitleTrKey?: string;
  modalContentTrKey?:
    | string
    | (
        | string
        | string[]
        | { trKey: string; values?: Record<string, ReactNode> }
      )[];
  ModalComponent?: React.FC;
  modalContent?: ReactNode;
  modalHtml?: string;
  modalHtmlTrValues?: Record<string, TranslationValue>;
  modalButtons?: {
    [key in ButtonTypes]?: {
      textTrKey: string;
      onClick?: () => void;
    };
  };
  modalType?: ModalTypes;
  maskClosable?: boolean;
}
