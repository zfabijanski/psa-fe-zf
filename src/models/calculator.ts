import {
  Frequency,
  InvestmentProfile,
  ProductCategory,
  Role,
} from "models/common";

export type CalculationState = {
  status?: CalculationStatus;
  calculationFeatureChanged: boolean;
};

export interface ICalculationDetailsDto extends ICalculationDto {
  calculation_id: number;
  working_date: string;
  premiums_total: number;
  guaranteed_death_benefit: number | null;
  guaranteed_maturity_benefit: number | null;
  projected_maturity_value_egr_m: number | null;
  product_categories: ProductCategory[];
}

export interface ICalculateResponseDto {
  calculation: ICalculationDetailsDto;
  msg: IMessage[];
}

interface IMessage {
  msg_code: string;
  msg_type: MessageType;
  msg_mesage: string;
  msg_function: string;
  msg_key?: string;
}

export enum MessageType {
  Ok = "OK",
  Error = "ERROR",
}

export interface ICalculationDto {
  uuid: string | null;
  version: string | null;
  source_system: string | null;
  product_guid: number;
  insureds_list: IInsured[];
  covers_list: ICover[];
  funds_list: IFund[];
}

export interface IInsured {
  insured_idx: number;
  insured_uuid: string | null;
  age_in_months: number | null;
  date_of_birth: string | null;
  role: Role;
}

export interface ICover {
  cover_id?: number;
  cover_idx: number;
  insured_idx: number;
  code: string;
  frequency_code: Frequency | null;
  payments: number | null;
  direction_of_calculation: Direction | null;
  investment_profile: InvestmentProfile | null;
  indexation: YesNo | null;
  sum_assured: number | null;
  premium_per_freq: number | null;
  duration: number | null;
  idx_option: YesNo | null;
  rop_option: YesNo | null;
  add_sum_assured: number | null;
  ending_age: number | null;
  exp_mat_ben: number | null;
}

export interface IFundGuarantee {
  gtee_code: string;
  allocation_percent: number;
}

export interface IFund {
  cover_idx: number;
  fund_code: string | null;
  fund_name?: string | null;
  fund_risk_profile: string | null;
  allocation_value: number | null;
  fund_gtees_list?: IFundGuarantee[] | null;
}

export enum Direction {
  PremiumToSumAssured = "PREM_TO_SA",
  SumAssuredToPremium = "SA_TO_PREM",
  MaturityBenefitToPremium = "MATBEN_TO_PREM",
}

export enum YesNo {
  Yes = "Y",
  No = "N",
}

export enum CalculationStatus {
  NotCalculated = "NOT_CALCULATED",
  Invalid = "INVALID",
  Calculated = "CALCULATED",
  Illustrated = "ILUSTRATED",
}
