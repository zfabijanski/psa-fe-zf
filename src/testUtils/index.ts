export * from "./TestProviders";
export * from "./mockCreators";
export * from "./helpers";
