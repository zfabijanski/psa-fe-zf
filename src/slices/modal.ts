import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { IModalData } from "models/Modal";

const initialState = {
  data: {} as IModalData,
};

export const modalSlice = createSlice({
  name: "modal",
  initialState,
  reducers: {
    showModal(state, action: PayloadAction<IModalData>) {
      state.data = action.payload;
    },
    closeModal(state) {
      state.data = {};
    },
  },
});

export const { showModal, closeModal } = modalSlice.actions;
