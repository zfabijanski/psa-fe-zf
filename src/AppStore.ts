import {
  configureStore,
  ThunkAction,
  ThunkDispatch,
  Action,
  ActionCreator,
  combineReducers,
} from "@reduxjs/toolkit";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";

// slices
import { authSlice } from "slices/auth";
import { auxiliaryCalculatorSlice } from "slices/auxiliaryCalculator";
import { bopDropdownListsSlice } from "slices/bopDropdownLists";
import { bopSlice, actions as bopActions } from "slices/bop";
import { businessWithPrudentialSlice } from "slices/businessWithPrudential";
import { calculatorSlice } from "slices/calculator";
import { confirmModalSlice } from "slices/confirmModal";
import { coversConfigSlice } from "slices/coversConfig";
import { dictionariesSlice } from "slices/dictionaries";
import { fullscreenSpinnerSlice } from "slices/fullscreenSpinner";
import { fundRiskProfilesConfigSlice } from "slices/fundRiskProfilesConfig";
import { iddSlice, actions as iddActions } from "slices/idd";
import { librarySlice } from "slices/library";
import { mailSlice } from "slices/mail";
import { meetingsSlice } from "slices/meetings";
import { modalSlice } from "slices/modal";
import { productCoversLimitsSlice } from "slices/productCoversLimits";
import { productsConfigSlice } from "slices/productsConfig";
import { productsSlice } from "slices/products";
import { quarterlyIncomeCalculatorSlice } from "slices/quarterlyIncomeCalculator";
import { translationsSlice } from "slices/translations";
import { illustrationsSlice } from "slices/illustrations";

export const rootReducer = combineReducers({
  auth: authSlice.reducer,
  auxiliaryCalculator: auxiliaryCalculatorSlice.reducer,
  bop: bopSlice.reducer,
  bopDropdownLists: bopDropdownListsSlice.reducer,
  businessWithPrudential: businessWithPrudentialSlice.reducer,
  calculator: calculatorSlice.reducer,
  confirmModal: confirmModalSlice.reducer,
  coversConfig: coversConfigSlice.reducer,
  dictionaries: dictionariesSlice.reducer,
  fullscreenSpinner: fullscreenSpinnerSlice.reducer,
  fundRiskProfilesConfig: fundRiskProfilesConfigSlice.reducer,
  idd: iddSlice.reducer,
  illustrations: illustrationsSlice.reducer,
  library: librarySlice.reducer,
  mail: mailSlice.reducer,
  meetings: meetingsSlice.reducer,
  modal: modalSlice.reducer,
  productCoversLimits: productCoversLimitsSlice.reducer,
  products: productsSlice.reducer,
  productsConfig: productsConfigSlice.reducer,
  quarterlyIncomeCalculator: quarterlyIncomeCalculatorSlice.reducer,
  translations: translationsSlice.reducer,
});

const devToolsActionCreators = {
  auth: authSlice.actions,
  auxiliaryCalculator: auxiliaryCalculatorSlice.actions,
  bop: bopActions,
  bopDropdownLists: bopDropdownListsSlice.actions,
  businessWithPrudential: businessWithPrudentialSlice.actions,
  calculator: calculatorSlice.actions,
  confirmModal: confirmModalSlice.actions,
  coversConfig: coversConfigSlice.actions,
  dictionaries: dictionariesSlice.actions,
  fullscreenSpinner: fullscreenSpinnerSlice.actions,
  fundRiskProfilesConfig: fundRiskProfilesConfigSlice.actions,
  idd: iddActions,
  illustrations: illustrationsSlice.actions,
  library: librarySlice.actions,
  mail: mailSlice.actions,
  meetings: meetingsSlice.actions,
  modal: modalSlice.actions,
  productCoversLimits: productCoversLimitsSlice.actions,
  products: productsSlice.actions,
  productsConfig: productsConfigSlice.actions,
  quarterlyIncomeCalculator: quarterlyIncomeCalculatorSlice.actions,
  translations: translationsSlice.actions,
  // Workaround for the devTools middleware not being able to handle
  // action creators that return a function
} as unknown as Record<string, ActionCreator<unknown>>;

const store = configureStore({
  reducer: rootReducer,
  devTools: {
    name: `PSA Online, timestamp:${Date.now()}`,
    actionCreators: devToolsActionCreators,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type RootState = Readonly<ReturnType<typeof rootReducer>>;
export type ThunkResult<T = void> = ThunkAction<
  T,
  RootState,
  undefined,
  Action<any>
>;

export type TAppDispatch = ThunkDispatch<RootState, any, Action>;
export const useAppDispatch = () => useDispatch<TAppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export { configureStore };

export default store;
