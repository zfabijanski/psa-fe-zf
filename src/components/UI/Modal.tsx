import { FC } from "react";
import styled, { createGlobalStyle, css } from "styled-components/macro";
import { FocusOn } from "react-focus-on";
import { useAppDispatch, useAppSelector } from "AppStore";
import { Icon } from "./Icon";
import { TypoH2, TypoBodyRegular } from "./typography";
import PruButton from "./PruButton/PruButton";
import { FormattedMessage, useIntl } from "react-intl";
import { closeModal } from "slices/modal";
import { ModalTypes } from "../../models/Modal";

const HideBodyOverflow = createGlobalStyle`
  body {
    overflow: hidden;
  }
`;

export const Modal: FC = () => {
  const intl = useIntl();
  const dispatch = useAppDispatch();
  const { data } = useAppSelector((state) => state.modal);
  const {
    modalIllustration,
    modalTitleTrKey,
    modalContentTrKey,
    modalContent,
    modalButtons,
    ModalComponent,
    modalHtml,
    modalHtmlTrValues,
    modalType,
    maskClosable = true,
  } = data;
  const { cancel, confirm } = modalButtons || {
    confirm: {
      textTrKey: "modal.ok",
    },
  };

  const showModal = !!Object.keys(data).length;

  if (!showModal) return <div />;

  return (
    <FocusOn>
      <Mask onClick={maskClosable ? () => dispatch(closeModal()) : undefined}>
        <HideBodyOverflow />
        <Wrapper onClick={(e) => e.stopPropagation()} role="dialog">
          <ModalContent>
            {modalIllustration && (
              <StyledIcon name={modalIllustration} width={120} />
            )}
            {modalTitleTrKey && (
              <StyledTypoH2 modalType={modalType}>
                <FormattedMessage id={modalTitleTrKey} />
              </StyledTypoH2>
            )}
            {modalContentTrKey &&
              (Array.isArray(modalContentTrKey) ? (
                modalContentTrKey.map((tr, index) => (
                  <SingleMessage
                    key={index}
                    notLast={
                      !!(
                        modalContentTrKey &&
                        index < modalContentTrKey.length - 1
                      )
                    }
                  >
                    {Array.isArray(tr) ? (
                      tr.map((singleTr, index) => (
                        <FormattedMessage key={index} id={singleTr} />
                      ))
                    ) : typeof tr === "string" ? (
                      <FormattedMessage id={tr} />
                    ) : (
                      <FormattedMessage id={tr.trKey} values={tr.values} />
                    )}
                  </SingleMessage>
                ))
              ) : (
                <FormattedMessage id={modalContentTrKey} />
              ))}
            {modalContent && modalContent}
            {ModalComponent && <ModalComponent />}
            {modalHtml && (
              <ModalHtml
                dangerouslySetInnerHTML={{
                  __html: intl.formatMessage(
                    { id: modalHtml },
                    modalHtmlTrValues
                  ) as string,
                }}
              />
            )}
          </ModalContent>

          <Separator />

          <ButtonsContainer>
            {cancel && (
              <StyledButton
                onClick={(e) => {
                  e.stopPropagation();
                  dispatch(closeModal());
                  cancel.onClick && cancel.onClick();
                }}
                buttonType="secondary"
                size="large"
              >
                <FormattedMessage id={cancel.textTrKey} />
              </StyledButton>
            )}

            {confirm && (
              <PruButton
                onClick={(e) => {
                  e.stopPropagation();
                  dispatch(closeModal());
                  confirm.onClick && confirm.onClick();
                }}
                size="large"
                buttonType={
                  modalType === ModalTypes.danger ? "error" : "primary"
                }
              >
                <FormattedMessage
                  defaultMessage={"OK"}
                  id={confirm.textTrKey}
                />
              </PruButton>
            )}
          </ButtonsContainer>
        </Wrapper>
      </Mask>
    </FocusOn>
  );
};

const SingleMessage = styled.div<{
  notLast?: boolean;
}>`
  ${({ notLast }) =>
    notLast &&
    css`
      margin-bottom: 10px;
    `}
`;

const Mask = styled.div.attrs({
  "data-testid": "modal-mask",
})`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 100;
  padding: 24px;
  display: flex;
  align-items: center;
  justify-content: center;

  &:before {
    content: "";
    z-index: -1;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    opacity: 0.6;
    background-color: ${({ theme }) => theme.newColors.gray100};
  }
`;

const Wrapper = styled.div`
  background-color: ${({ theme }) => theme.newColors.gray10};
  max-width: 640px;
  margin-top: -100px;
  width: 100%;
  display: block;
  box-shadow: ${({ theme }) => theme.shadows.elevation2};
`;

const ModalContent = styled(TypoBodyRegular)`
  padding: 32px 24px;
  max-height: calc(100vh - 250px);
  white-space: break-spaces;
  text-align: center;
  flex-direction: column;
  word-wrap: break-word;
  overflow: auto;

  ${({ theme }) =>
    css`
      @media (max-width: ${theme.breakpoints.lg}px) {
        max-height: calc(100vh - 250px - 50px);
      }
    `}
`;

const ModalHtml = styled.div`
  white-space: pre-wrap;
`;

const StyledIcon = styled(Icon)`
  margin-bottom: 24px;
  display: flex;
  justify-content: center;
`;

const StyledTypoH2 = styled(TypoH2)<{ modalType?: ModalTypes }>`
  color: ${({ theme, modalType }) =>
    modalType === ModalTypes.danger
      ? theme.newColors.error
      : theme.newColors.primary100};
  margin-bottom: 24px;
`;

const Separator = styled.div`
  height: 1px;
  background-color: ${({ theme }) => theme.newColors.gray30};
`;

const ButtonsContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 16px 0;
`;

const StyledButton = styled(PruButton)`
  margin-right: 16px;
`;
