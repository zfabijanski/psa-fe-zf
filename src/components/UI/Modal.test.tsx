import { Modal } from "./Modal";
import { ConfigureStoreOptions } from "@reduxjs/toolkit";
import { createModalData, render, screen, fireEvent } from "testUtils";
import { configureStore, rootReducer as reducer } from "AppStore";
import { showModal } from "slices/modal";
import { ModalTypes } from "models/Modal";

const middleware: ConfigureStoreOptions["middleware"] = (
  getDefaultMiddleware
) =>
  getDefaultMiddleware({
    serializableCheck: false,
  });

let store = configureStore({
  reducer,
  middleware,
});

const setup = () => render(<Modal />, {}, { store, renderModal: false });

describe("Modal", () => {
  beforeEach(() => {
    store = configureStore({ reducer, middleware });
  });

  it("should render the component", () => {
    store.dispatch(showModal(createModalData()));
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("should close modal when clicking on mask", async () => {
    store.dispatch(showModal(createModalData()));
    setup();
    fireEvent.click(screen.getByTestId("modal-mask"));
    expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
  });

  it("shouldn't close modal when clicking on mask when maskClosable is false", () => {
    store.dispatch(
      showModal(
        createModalData({
          maskClosable: false,
        })
      )
    );
    setup();
    fireEvent.click(screen.getByTestId("modal-mask"));
    expect(screen.getByRole("dialog")).toBeInTheDocument();
  });

  it("shouldn't close modal when clicking on modal", () => {
    store.dispatch(showModal(createModalData()));
    setup();
    fireEvent.click(screen.getByRole("dialog"));
    expect(screen.getByRole("dialog")).toBeInTheDocument();
  });

  it("should handle nested array of modalContentTrKeys", () => {
    store.dispatch(
      showModal(
        createModalData({
          modalContentTrKey: ["modalContentTrKey1", ["modalContentTrKey2"]],
        })
      )
    );
    setup();
    expect(screen.getByText("modalContentTrKey1")).toBeInTheDocument();
    expect(screen.getByText("modalContentTrKey2")).toBeInTheDocument();
  });

  it("should render custom modal component", () => {
    const CustomComponent = () => <div>CustomComponent</div>;
    store.dispatch(
      showModal(
        createModalData({
          ModalComponent: CustomComponent,
        })
      )
    );
    setup();
    expect(screen.getByText("CustomComponent")).toBeInTheDocument();
  });

  it("should render modal with html content", () => {
    store.dispatch(
      showModal(
        createModalData({
          modalHtml: "<div>modalHtml</div>",
        })
      )
    );
    setup();
    expect(screen.getByText("modalHtml")).toBeInTheDocument();
  });

  it("should show cancel button and close modal when clicked", () => {
    const onCancel = jest.fn();

    store.dispatch(
      showModal(
        createModalData({
          modalButtons: {
            confirm: { textTrKey: "OK" },
            cancel: { textTrKey: "cancel", onClick: onCancel },
          },
        })
      )
    );
    setup();
    fireEvent.click(screen.getByText("cancel"));
    expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
    expect(onCancel).toHaveBeenCalledTimes(1);
  });

  it("should show confirm button and close modal when clicked", () => {
    const onConfirm = jest.fn();

    store.dispatch(
      showModal(
        createModalData({
          modalButtons: {
            confirm: { textTrKey: "OK", onClick: onConfirm },
            cancel: { textTrKey: "cancel" },
          },
        })
      )
    );
    setup();
    fireEvent.click(screen.getByText("OK"));
    expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
    expect(onConfirm).toHaveBeenCalledTimes(1);
  });

  it("should render error modal type", () => {
    store.dispatch(
      showModal(
        createModalData({
          modalType: ModalTypes.danger,
        })
      )
    );
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
