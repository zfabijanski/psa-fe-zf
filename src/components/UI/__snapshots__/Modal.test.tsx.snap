// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render error modal type 1`] = `
.c4 {
  display: inline-block;
  cursor: inherit;
}

.c4 > svg {
  width: 120px;
  height: 100%;
  display: block;
}

.c6 {
  font-weight: 600;
  font-size: 32px;
  line-height: 40px;
}

.c2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 20px;
}

.c10 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #FFFFFF;
  background-color: #D13C15;
  border: 2px solid #D13C15;
}

.c10 svg g,
.c10 svg g path,
.c10 svg g circle,
.c10 svg g rect {
  stroke: #FFFFFF;
}

.c10 svg g rect {
  fill: #FFFFFF;
}

.c10 svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c10 svg > path {
  stroke: #FFFFFF;
}

.c10:hover {
  color: #FFFFFF;
  background-color: #D13C15;
  border: 2px solid #D13C15;
}

.c10:hover svg g,
.c10:hover svg g path,
.c10:hover svg g circle,
.c10:hover svg g rect {
  stroke: #FFFFFF;
}

.c10:hover svg g rect {
  fill: #FFFFFF;
}

.c10:hover svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c10:hover svg > path {
  stroke: #FFFFFF;
}

.c10:disabled {
  cursor: initial;
  color: #FFFFFF;
  background-color: #D13C15;
  border: 2px solid #D13C15;
}

.c10:disabled svg g,
.c10:disabled svg g path,
.c10:disabled svg g circle,
.c10:disabled svg g rect {
  stroke: #FFFFFF;
}

.c10:disabled svg g rect {
  fill: #FFFFFF;
}

.c10:disabled svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c10:disabled svg > path {
  stroke: #FFFFFF;
}

.c10:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.c0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 100;
  padding: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0:before {
  content: "";
  z-index: -1;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0.6;
  background-color: #2F2F2F;
}

.c1 {
  background-color: #F3F3F3;
  max-width: 640px;
  margin-top: -100px;
  width: 100%;
  display: block;
  box-shadow: 0px 1px 2px rgba(10,27,56,0.08),0px 2px 4px rgba(10,27,56,0.12);
}

.c3 {
  padding: 32px 24px;
  max-height: calc(100vh - 250px);
  white-space: break-spaces;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  word-wrap: break-word;
  overflow: auto;
}

.c5 {
  margin-bottom: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  color: #D13C15;
  margin-bottom: 24px;
}

.c8 {
  height: 1px;
  background-color: #E3E3E3;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 16px 0;
}

@media (max-width:1199px) {
  .c3 {
    max-height: calc(100vh - 250px - 50px);
  }
}

<div>
  <div
    aria-hidden="true"
    data-focus-guard="true"
    data-focus-on-hidden="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="c0"
      data-testid="modal-mask"
    >
      <div
        class="c1"
        role="dialog"
      >
        <div
          class="c2 c3"
        >
          <div
            class="c4 c5"
          >
            <svg>
              alert-fill.svg
            </svg>
          </div>
          <div
            class="c6 c7"
          >
            modalTitleTrKey
          </div>
          modalContentTrKey
          modalContent
        </div>
        <div
          class="c8"
        />
        <div
          class="c9"
        >
          <button
            class="c10"
            type="button"
          >
            OK
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-focus-guard="true"
    data-focus-on-hidden="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;

exports[`Modal should render the component 1`] = `
.c4 {
  display: inline-block;
  cursor: inherit;
}

.c4 > svg {
  width: 120px;
  height: 100%;
  display: block;
}

.c6 {
  font-weight: 600;
  font-size: 32px;
  line-height: 40px;
}

.c2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 20px;
}

.c10 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #FFFFFF;
  background-color: #055A60;
  border: 2px solid #055A60;
}

.c10 svg g,
.c10 svg g path,
.c10 svg g circle,
.c10 svg g rect {
  stroke: #FFFFFF;
}

.c10 svg g rect {
  fill: #FFFFFF;
}

.c10 svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c10 svg > path {
  stroke: #FFFFFF;
}

.c10:hover {
  color: #FFFFFF;
  background-color: #181818;
  border: 2px solid #181818;
}

.c10:hover svg g,
.c10:hover svg g path,
.c10:hover svg g circle,
.c10:hover svg g rect {
  stroke: #FFFFFF;
}

.c10:hover svg g rect {
  fill: #FFFFFF;
}

.c10:hover svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c10:hover svg > path {
  stroke: #FFFFFF;
}

.c10:disabled {
  cursor: initial;
  color: #FFFFFF;
  background-color: #9BBDBF;
  border: 2px solid #9BBDBF;
}

.c10:disabled svg g,
.c10:disabled svg g path,
.c10:disabled svg g circle,
.c10:disabled svg g rect {
  stroke: #FFFFFF;
}

.c10:disabled svg g rect {
  fill: #FFFFFF;
}

.c10:disabled svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c10:disabled svg > path {
  stroke: #FFFFFF;
}

.c10:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.c0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 100;
  padding: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0:before {
  content: "";
  z-index: -1;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0.6;
  background-color: #2F2F2F;
}

.c1 {
  background-color: #F3F3F3;
  max-width: 640px;
  margin-top: -100px;
  width: 100%;
  display: block;
  box-shadow: 0px 1px 2px rgba(10,27,56,0.08),0px 2px 4px rgba(10,27,56,0.12);
}

.c3 {
  padding: 32px 24px;
  max-height: calc(100vh - 250px);
  white-space: break-spaces;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  word-wrap: break-word;
  overflow: auto;
}

.c5 {
  margin-bottom: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  color: #055A60;
  margin-bottom: 24px;
}

.c8 {
  height: 1px;
  background-color: #E3E3E3;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 16px 0;
}

@media (max-width:1199px) {
  .c3 {
    max-height: calc(100vh - 250px - 50px);
  }
}

<div>
  <div
    aria-hidden="true"
    data-focus-guard="true"
    data-focus-on-hidden="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="c0"
      data-testid="modal-mask"
    >
      <div
        class="c1"
        role="dialog"
      >
        <div
          class="c2 c3"
        >
          <div
            class="c4 c5"
          >
            <svg>
              alert-fill.svg
            </svg>
          </div>
          <div
            class="c6 c7"
          >
            modalTitleTrKey
          </div>
          modalContentTrKey
          modalContent
        </div>
        <div
          class="c8"
        />
        <div
          class="c9"
        >
          <button
            class="c10"
            type="button"
          >
            OK
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-focus-guard="true"
    data-focus-on-hidden="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;
